commands/files_cleanup.ts

```diff
@@ -343,17 +369,21 @@ export default class CleanupFiles extends BaseCommand {
           .whereIn(column, toNull)
           .update(values)
           .exec();
-        // Remove from the FileEntry table
-        await FileEntry.query().delete().whereIn("id", toNull).exec();
         this.logger.info(
-          `Set ${toNull.length} references to null for ${collection.relationData.model.name}. Removed all from FileEntry table.`,
+          `Set ${toNull.length} references to null for ${collection.relationData.model.name}`,
         );
+        return toNull;
       }),
+    ).then((values) => values.filter((value) => value !== null).flat());
+    // Remove from the FileEntry table
+    await FileEntry.query().delete().whereIn("id", removedIds).exec();
```

Comment:
this might still fail if the same file is used in two models, where the field in one is nullable and the other is not